src/vaults/SectorCrossVault.sol

```diff
@@ -206,15 +190,79 @@ contract SectorCrossVault is SectorBase {
 		}
 	}
 
+	// Do linear search on vaultList -> O(n)
+	function removeVault(address _vault) external onlyOwner {
+		addrBook[_vault].allowed = false;
+
+		uint256 length = vaultList.length;
+		for (uint256 i = 0; i < length; ) {
+			if (vaultList[i] == _vault) {
+				vaultList[i] = vaultList[length - 1];
+				vaultList.pop();
+				return;
+			}
+			unchecked {
+				i++;
+			}
+		}
+	}
+
 	/*/////////////////////////////////////////////////////
-						Modifiers
+							Overrides
 	/////////////////////////////////////////////////////*/
 
-	modifier harvestLock() {
-		if (harvestLedger.count != 0) revert OnGoingHarvest();
-		_;
+	function addVault(
+		address _vault,
+		uint16 _chainId,
+		uint16 _srcPostmanId,
+		uint16 _dstPostmanId,
+		bool _allowed
+	) external override onlyOwner {
+		_addVault(_vault, _chainId, _srcPostmanId, _dstPostmanId, _allowed);
+		vaultList.push(_vault);
 	}
 
+	function setMessageActionCallback() external override onlyOwner {
```

Comment:
do we need the as a method?
should this be done in the constructor?
We can also potentially hardcode this in XChainIntegrator and have the vaults override the handlers... this might be a bit simpler

Reply:
Agree, for some reason I thought this wouldn't work but it does work. Much better this way.